### PR TITLE
Artifact manager handle closing of classloader before deleting direct…

### PR DESCRIPTION
**Pull request by gsps1**
…ory contents

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/DefaultArtifactManager.java [hunk 1]
```diff
@@ -179,25 +180,31 @@ private CloseableClassLoader createAndGetClassLoader(HttpRequest httpRequest,
...
+        new DirectoryClassLoader(unpackedDir, parentClassLoader == null ? bootstrapClassLoader : parentClassLoader);
+      return new CloseableClassLoader(directoryClassLoader, new ClassLoaderCleanup(directoryClassLoader, unpackedDir));
     } else {
       throw new IOException(String.format("Exception while getting artifacts list %s",
                                           httpResponse.getResponseBodyAsString()));
     }
   }
 
-  class DeleteContents implements Closeable {
+  class ClassLoaderCleanup implements Closeable {
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/DefaultArtifactManager.java]: `private static final`

**Comment by chtyim** [comment]
Just one comment, reset LGTM.

**Comment by chtyim** [comment]
LGTM.

**Comment by gsps1** [comment]
https://builds.cask.co/browse/CDAP-RUT977-4 build green, rebased commits, merging.

**Pull request merged**